metrics/web/docs/features/firestore_emulator_for_driver_tests/01_firestore_emulator_for_driver_tests.md

```diff
@@ -0,0 +1,232 @@
+
+```dart
+setupAll(() {
+    Firestore.instance.settings(
+        host: 'localhost:8080',
+        sslEnabled: false,
+        persistenceEnabled: false,
+    );
+});
+```
+
+The specified port in the `host` argument must be equal to the emulator's port.
+
+# Design
+
+### Architecture
+
+> Fundamental structures of the feature and context (diagram).
+
+![Firestore emulator class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/firestore_emulator_design/metrics/web/docs/features/firestore_emulator_for_driver_tests/diagrams/firestore_emulator_class_diagram.puml)
+
+### User Interface
+
+> How users will interact with the feature (API, CLI, Graphical interface, etc.).
+
+The feature is introduced by adding additional parameters to the integration tests running command.
+
+The next table contains the definitions of the parameters with their default values.
+
+| Parameter         | Default | Description                                                           |
+| ----------------- | -------- | --------------------------------------------------------------------- |
+| **use-firestore-emulator**  | true     | Determines if the integration tests are using the `Firebase Emulator`. |
+| **emulator-port** | 8080     | Specifies the emulator's running port.                                  |
+
+The following code snippet shows an example of using the new parameters:
+
+```bash
+dart test_driver/main.dart --use-firestore-emulator --emulator-port=8081
+```
+
+Or, if you accept the defaults, you can omit them.
+
+As described in the [Prototyping](#prototyping) section, we should run integration tests once the `Firebase Emulator` started.
+
+### Database
+
+> How relevant data will be persisted.
+
+Integration tests interact with the application that uses data from the database. As we don't want to use the production environment, we should use the `Firebase local emulator`.
+
+By default, the emulator starts with no data, so we should add test data used during testing. The best approach to do so is importing the test data to the Firestore emulator once running. But before exporting the test data, we should create it. The `emulators:export` command allows to export the data from the running Firestore emulator. 
+
+So, to create test data for the Firestore emulator, we should follow the next steps: 
+
+1. Run the Firestore emulator.
+2. Create test data in the Firestore emulator using the Firestore emulator UI. 
+3. Run the `firebase emulators:export export_directory` command to export test data to the `export_directory`.
+
+_**Note:** The `export_directory` in our case should be a `firebase/tests/firestore/test_data`._
+
+Once we have exported data from the Firestore emulator, we can run an emulator with the exported data using the following command:
+
+```bash
+firebase emulators:start --import=export_directory
```

Comment:
It would be great to have a diagram showing the main classes and relationships used to run such commands. Also, It would be useful to have a `sequence` diagram showing the process of running driver tests using the Firestore emulator.